### ISSUE TITLE
System eShop application crash with EUR <v10.4.

### ISSUE THREAD

**Issue by ghost** [yellows8/ctr-httpwn#5]
Hi, I am using a 2DS on fw 9.9 E. After using ctr httpwn I can play online and use miiverse but after eshop login my 2DS freezes everytime and need a hardreset. Both homemenu and hans have the same issue. Help please.

**Comment by ghost** [comment]
The 2DS freezes seconds after sucessful login with NNID on eshop main menu.

**Comment by yellows8** [comment]
"freezes"?

**Comment by kymatx** [comment]
The same happens with my new 3ds xl on firmware 10.3.0.27E. After login the 3ds stops working and I have to reboot it.

**Comment by yellows8** [comment]
Video?

**Comment by kymatx** [comment]
https://youtu.be/QqOJbEewKhQ

**Comment by yellows8** [comment]
https://github.com/yellows8/ctr-httpwn#usage "For everything else, and for using system-eShop-app on >=v10.0.0-X, you can just return to Home Menu from hbmenu."

**Comment by kymatx** [comment]
I tried that too but I get the same error when I start the  shop from the Home Menu. I can show you a video later.

**Comment by kymatx** [comment]
https://youtu.be/Kmoz7F9RznM

**Comment by yellows8** [comment]
Ugh... FWIW I only tested ctr-httpwn on v9.6, v10.5, and {latest sysver}... :-/

**Comment by ghost** [comment]
The same on my system like on this videos by kymatx.

**Comment by Joba87** [comment]
Happens to me too :/ 10.3 EUR n3DS

**Comment by yellows8** [comment]
Can't repro with JPN New3DS v10.3.

**Comment by ghost** [comment]
Is this an EU fw problem only.

**Comment by NyanKats** [comment]
Same issue on EUR n3DS on 10.3 using v1, with the new 1.0.1 though I'm getting the "An error has occurred forcing the software to close" message and it then proceed to reboot.

On the other hand everything else works perfectly, I had an update available for one game and the eShop opened and updated it without issues, it seems to be the main page that's causing us to crash.

Thanks a lot for your work, really appreciate it.

**Comment by yellows8** [comment]
Updated: https://github.com/yellows8/ctr-httpwn/blob/master/README.md#supported-sysmodule-versions

**Comment by profi200** [comment]
Just to add my mustard:

I was ableto reproduce it the exact way as in the videos on N3DS 9.9. Crash/cmd logs are there and it's been worked on.

**Comment by yellows8** [comment]
Can someone test using this eShop QR-code on EUR <v10.4?

![tmp](https://cloud.githubusercontent.com/assets/585494/14061117/5904070e-f34e-11e5-8bf7-aa6bc2eea357.png)

**Comment by Joba87** [comment]
It just loads forever on the eShop loading screen for me. Then after a while it gives error: 005-4320

n3DS 10.3 EUR

Btw when I ran httpwn just now, the server was down or something so it used the cached version on SD card instead. Just saying if that changes anything.

**Comment by yellows8** [comment]
You _must_ use v1.0.1. https://github.com/yellows8/ctr-httpwn/releases

**Comment by Joba87** [comment]
Ahh, my apologies. The QR code takes me to Mutant Mudds Super Challenge's page, but it freezes after half a second like in a the video and then eshop crashes :/

**Comment by supermin77** [comment]
i tested the QR code in a 2DS 9.9 , " an error ocurred 007-2404" then back to e-shop and crash.

**Comment by gnmmarechal** [comment]
That is because the eShop app is outdated on those firmwares. You CAN install the newer eShop (from 10.4) CIA using a CIA Manager such as NASA. Make sure you install the CIA to NAND.
Essentially this: https://www.reddit.com/r/3dshacks/comments/476wy5/tutorialeshop_and_online_gameplay_on_outdated/

**Comment by NyanKats** [comment]
Same as Joba for me when trying to access using the QR code sys eShop freeze then crash.

**Comment by gnmmarechal** [comment]
@NyanKats that's because your eShop app is outdated. Update the eShop application with NASA.

**Comment by gnmmarechal** [comment]
This isn't an issue with httpwn.

**Comment by ghost** [comment]
@ gnmmarechal: Why are the friends and miiverse modules working fine with httpwn? As I know the modules was updated by Nintendo too in later firmware updates. As I see only EU Firmware 3DS´s have that eShop issue. Why? So only the eShop module have to be updated, right? So NVer and not Tiger?

**Comment by Joba87** [comment]
@gnmmarechal @yellows8 After trying to start up NASA for 2+ hours, it finally booted and I was able to update the eShop app to 10.7 ver and now everything works perfectly!

Thanks for the help and thanks for the great work yellows8. And sorry for the headache xD

**Comment by profi200** [comment]
That's not really how it is supposed to work. You should instead wait for a fix in HANS for this. Installing higher version titles **can** be problematic.

**Comment by gnmmarechal** [comment]
@the-checker 
the eShop app probably had an important update. You don't get any error message, so we know the update to the eShop is something that (probably not intentionally) disrupts the older eShop. You're wrong though. NVer isn't the eShop app. Tiger and Mint, update those. Mint isn't required, but I heard it's necessary to have it on latest version in order to add funds (and buying stuff).
@yellows8 

Could you add to the Readme that the fact that this doesn't work with 10.4 is not an issue with ctr-httpwn itself? And could you add that it can be bypassed by installing the latest eShop CIA (using NASA, for example), as stated in my tutorial for bypassing online restrictions for outdated firms?

And I guess it'd be fine to close the issue, it's not really an issue with httpwn, is it? Back when 10.4 was released, HANS + NVer + friends-module spoofing allowed me to log in to eShop, only for it to freeze. This is related to the way eShop works, not httpwn.

**Comment by ghost** [comment]
like profi200 said: "That's not really how it is supposed to work. You should instead wait for a fix in HANS for this. Installing higher version titles CAN be problematic." I will wait for smea and yellows8. If its not working then I have to update the 2DS.

**Comment by profi200** [comment]
After writing that i realized it is not HANS related because it happens with vanilla eShop too. If it is related to changes on Nintendos servers there is not much one can do. Otherwise it can probably be fixed with a patch using HANS.

**Comment by gnmmarechal** [comment]
@profi200 yep it happened before when 10.4 was released. It's probably a change on Nintendo's side, I'd say, because I changed no files and it stopped working once the newest eShop app was released. Since eShop thought I was on the latest fw, but it crashed on start. They probably changed how some important stuff works, and that rendered the older eShop unusable.

@the-checker Never had a problem with installing the newest NVer/friends-module/eShop/mint.

**Comment by yellows8** [comment]
Has anyone _ever_ had any issues with manual partial-sysupdates on physnand once installed? It could be done with ctr-httpwn itself as documented in the README, not sure whether to actually implement it for this though...

And yeah it's just the usual client<>server web API changes which aren't compatible with older client(system eShop app) versions(hence why HANS-eShop broke around when v10.0 was released until a workaround was implemented).

**Comment by gnmmarechal** [comment]
@yellows8 I've been doing partial system updates since 10.4 was released, works perfectly. Also, on the tutorial I wrote on Reddit, nobody had issues besides those who couldn't manage to boot NASA or FBI. I don't think httpwn should take care of partial sysupdating though.

**Comment by yellows8** [comment]
"I don't think httpwn should take care of partial sysupdating though." Why not?

**Comment by gnmmarechal** [comment]
@yellows8 well. It's my personal preference, I'd rather do it myself. Plus Memchunkhax2 would make the app have a lower bootrate, at least that's the case with FBI/sysUpdater/NASA. Anyway, FBI/NASA force you to shutdown since they'll crash upon exit. I don't know what causes this, but would it be because of mh2, I suppose? That'd probably be an issue, no? Since this isn't a permanent patch. Hum... Although it'd be nice to have it optional. And what about making this automatically patch stuff on boot and then exit so that we could boot to it directly from Menuhax and it'd patch the system on startup then continue to sysNAND? Then we'd use a different entrypoint to boot into HBL. I guess Menuhax would have to boot to a file different than boot.3dsx.  Well, it shouldn't be hard to implement, I suppose?

**Comment by yellows8** [comment]
I don't think you get it... ctr-httpwn doesn't need any arm11kernel-hax for doing this, just return SOAP data with the required titles via NetUpdateSOAP.php(_cough_ README _cough_).

The release builds of ctr-httpwn aren't really usable as a boot.3dsx right now.

**Comment by gnmmarechal** [comment]
@yellows8  wouldn't 10.3 or lower users need to install the latest eShop CIA? What do you mean by partial sysupdate? Isn't that it?

**Comment by yellows8** [comment]
Sure but you don't need to use any CIA-installers to do so.

**Comment by gnmmarechal** [comment]
@yellows8 but, you need memchunkhax2, no? to do that?

**Comment by yellows8** [comment]
No......_README_

**Comment by gnmmarechal** [comment]
@yellows8 I might be misunderstanding or misreading it, but where do you say in the readme that?

**Comment by ABDOULemul** [comment]
Same here with N3DS ofw 9.9.0_26E and Menuhax exploit. Eshop freeze when logging INN successfully.

**Comment by gnmmarechal** [comment]
@ABDOULemul update the eShop CIA with the latest eShop CIA. Check the title and version of the title from the update report for your region and download it using 3DNUS.

**Comment by yellows8** [comment]
https://github.com/yellows8/ctr-httpwn#exploit-details "NIM: ..."

**Comment by gnmmarechal** [comment]
@yellows8 oh, I had skipped that, never mind. In that case, it'd be nice, but I'd rather have it optional.

**Comment by NyanKats** [comment]
Thanks gnmmarechal! Though I'd rather avoid that kind of solution, if it wasn't for potential bricks I'd be downgrading instead but I value way too much being able to use emulators. The risk of soft bricks leading to update, and my lack of entrypoints as I doesn't own a copy of Cubic Ninja or Oot simply isn't worth it.

I like yellows8 suggestion of having httpwn take care of it on the other hand, it seems safer.

**Comment by yellows8** [comment]
Not sure if/when I'll even start implementing that FWIW.

**Comment by kymatx** [comment]
I just found out that the eshop works fine on 10.3 when I use qr codes. I tried it with the qr code for hyrule warriors and pokedex. I didn't try to add funds yet but downloading demo versions works fine.
It seems everything works except the eshop main menu.

**Comment by Marsh92** [comment]
Hi I have a question about the hax payloads. How do you update from 2.5 to 2.7 so I can exit without rebooting? I'm currently using menuhax on 10.5 and nothing else with hax 2.5

**Comment by Marsh92** [comment]
Hi I have a question about the hax payloads. How do you update from 2.5 to 2.7 so I can exit without rebooting? I'm currently using menuhax on 10.5 and nothing else with hax 2.5

**Comment by Marsh92** [comment]
Sorry I just posted again on accident

**Comment by yellows8** [comment]
@Marsh92  Just reinstall menuhax.

**Comment by Marsh92** [comment]
That worked thank you very much :). Now I have another question. Do you press x to exit without rebooting? I read that on twitter and some people don't have that as an option.

**Comment by gnmmarechal** [comment]
@Marsh92 update the boot.3dsx on your SD Card with the latest version of either Homebrew Menu or Gridlauncher.

**Comment by Marsh92** [comment]
@gnmmarechal from the starter kit?

**Comment by gnmmarechal** [comment]
@Marsh92 yeah, that one will do.

**Comment by Marsh92** [comment]
@gnmmarechal okay that's all I needed to know thanks:)

**Comment by Marsh92** [comment]
When doing this do you first set your system date to 1-1-2000 like you do with that browser-version-check bypass or no?

**Comment by yellows8** [comment]
@Marsh92 https://github.com/yellows8/ctr-httpwn#usage

**Comment by Marsh92** [comment]
@yellows8 "when the system date is not set correctly(like with the browser-version-check bypass)." So yes?

**Comment by gnmmarechal** [comment]
@Marsh92 No. I have the date set correctly and use HTTPWN.

**Comment by gnmmarechal** [comment]
@Marsh92 " Do not use this when the system date is not set correctly"

**Comment by gnmmarechal** [comment]
more like so, no

**Comment by Marsh92** [comment]
@gnmmarechal so you just leave the date the way it is today?

**Comment by Marsh92** [comment]
@gnmmarechal no changing the date to 1-1-2000 at all?

**Comment by gnmmarechal** [comment]
@Marsh92 " Do not use this when the system date is not set correctly"
seriously.

**Comment by Marsh92** [comment]
@gnmmarechal ok and sorry I was just having a little bit of confusion on that part :)

**Comment by Marsh92** [comment]
@gnmmarechal I should've asked this before but just out of curiosity what date did you set your system on?

**Comment by gnmmarechal** [comment]
@Marsh92 hum... the right one

**Comment by Marsh92** [comment]
Which is?

**Comment by gnmmarechal** [comment]
@Marsh92 Literally - the one right now...atm 19:40, 22/04/2016....

**Comment by Marsh92** [comment]
@gnmmarechal ok that's what I have too. I pretty much got ctr-httpwn and updated the boot.3dsx as well. I'll finally try this out later since I'm a little busy right now. By the way to be honest with you I don't really care about the eshop as of right now, all that matters to me right now is the online access in games and pretty much getting the 20th anniversary jirachi from mystery gift on alpha sapphire as well as volcanion once I get the serial code. All that aside, thank you and yellows8 for helping me make sure I know what I'm doing when using this, if there's anything else I need help with I'll let either of you know :)

**Comment by gnmmarechal** [comment]
@Marsh92 if you're on o3DS, you won't be able to use httpwn for online in games like Smash and MH4U, but there are workarounds for that (aka frankenfirm, or just go and downgrade, I suppose)

**Comment by Marsh92** [comment]
Well I don't really have have smash or monster hunter 4 so I'm pretty much good there

**Comment by Marsh92** [comment]
@gnmmarechal I just tested httpwn now and it really worked on alpha sapphire online and now I have my 20th anniversary jirachi. Thanks again to you and yellows8

**Comment by ABDOULemul** [comment]
Thanks yellows8. It's working Well 😉

**Comment by mistaq** [comment]
I checked with a Demo link via Spotpass, and it worked fine. It seems like only the eShop home page crashes, while individual games/demos storefronts aren't. Searching also crashes the eShop.
